GlassFishORBHelper changed to singleton, lifecycle

dmatej opened the pull request:
- cherrypicked from #24616 + added two commits with code cleanup
- Original commit was created more than year ago, I think it was related to some race condition I noticed in logs, but now I am not sure. However I believe it is still bit better than older version. The CORBA/ORB initialization is a horrible thing - maybe it would be easier to rewrite it instead of refactoring. But that would need to know every piece of it first ...

dmatej committed:
Quick code cleanup

- I would like to do more, but it has too many open ends connected even to
  different libraries like glassfish-corba-orb

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

avpinchuk commented on appserver/orb/orb-connector/src/main/java/org/glassfish/enterprise/iiop/api/GlassFishORBHelper.java:
```diff
@@ -112,55 +103,54 @@ public ORB getORB() {
-
-                        Properties props = new Properties();
-                        props.setProperty(GlassFishORBFactory.ENV_IS_SERVER_PROPERTY, Boolean.valueOf(isServer).toString());
+        if (orb != null || destroyed) {
```
A small note on optimization ))

We can read volatile `orb` field into a local variable and work with it. This will reduce number of the volatile reads.

dmatej replied:
As it is outside the sync block it could even change between usages. It is not much probable here, but it was still a mistake. I have extracted the initProtocolManager method with the ORB parameter too.

dmatej committed:
GlassFishORBHelper changed to singleton, lifecycle

- Hard to trace all paths, but it seems this behaves better
- Using annotations instead of implements

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

# Conflicts:
#	appserver/orb/orb-connector/src/main/java/org/glassfish/enterprise/iiop/api/GlassFishORBHelper.java

dmatej committed:
Using the same orb instance outside the synchronized block, extracted method.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

avpinchuk commented on appserver/orb/orb-connector/src/main/java/org/glassfish/enterprise/iiop/api/GlassFishORBHelper.java:
```diff
@@ -108,55 +103,58 @@ public ORB getORB() {
+                    return orb;
                 }
+                protocolManager = initProtocolManager(orb);
+                return orb;
```
Maybe simplify this?

```java
if (isServer && protocolManager == null) {
    protocolManager = initProtocolManager(orb);
}
return orb;
```

dmatej replied:
Right!

dmatej committed:
Simplified if in GlassFishORBHelper

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>
